assets/app/views/browse/edit-build-config.html

```diff
@@ -0,0 +1,257 @@
+                                id="sourceContextDir"
+                                name="sourceContextDir"
+                                type="text"
+                                ng-model="updatedBuildConfig.spec.source.git.contextDir"
+                                aria-describedby="from_source_help"
+                                placeholder="/">
+                            </div>
+                          </div>
+                        </div>
+
+                        <div class="form-group">
+                          <div class="checkbox" ng-if="buildConfig.spec.strategy.type === 'Docker'">
+                            <label>
+                              <input type="checkbox" ng-model="options.noCache"/>
+                              Execute the docker build with the --no-cache=true flag
+                            </label>
+                          </div>
+                        </div>
+
+                        <div class="form-group">
+                          <label for="buildFrom">Build From: </label>
+                          <select 
+                            class="form-control simpleselector"
+                            ng-model="options.pickedBuildFromType"
+                            ng-options="type.id as type.title for type in buildFromTypes">
+                          </select>
+                        </div>
+
+                        <div ng-if="options.pickedBuildFromType=='ImageStreamTag'" class="form-group">
+                          <label for="builderImage">Builder Image</label>
+                          <div class="builder-image-selects multiselectors">
+                            <div class="label-with-selector">
+                              <label class="multiselector-label">Namespace</label>
+                              <select 
+                                class="form-control multiselector select-builder-namespace"
+                                ng-model="options.pickedBuildFromNamespace"
+                                ng-options="name for name in buildFrom.projects track by name"
+                                ng-change="updateImageStreams(options.pickedBuildFromNamespace, true, false)">
+                              </select>
+                            </div>
+                            <span class="punctuation">/</span>
+                            <div class="label-with-selector">
+                              <label class="multiselector-label">Image Stream</label>
+                              <select
+                                class="form-control multiselector select-builder-name"
+                                ng-model="options.pickedBuildFromImageStream"
+                                ng-options="name as name for name in buildFrom.imageStreams track by name"
+                                ng-change="updateTags(options.pickedBuildFromImageStream, true)">
+                              </select>
```

Comment:
I thought agreed you could type it in, somehow, whether its in those inputs
or a separate input.  We can't break if someone creates a build config from
the CLI that points to an IS tag that doesn't yet exist, and then goes and
edits that BC in the web console.

On Tue, Jan 5, 2016 at 1:16 PM, Sam Padgett <notifications@github.com>
wrote:

> In assets/app/views/browse/edit-build-config.html
> <https://github.com/openshift/origin/pull/6528#discussion_r48876185>:
>
> > +                              <select
> > +                                class="form-control multiselector select-builder-namespace"
> > +                                ng-model="options.pickedBuildFromNamespace"
> > +                                ng-options="name for name in buildFrom.projects track by name"
> > +                                ng-change="updateImageStreams(options.pickedBuildFromNamespace, true, false)">
> > +                              </select>
> > +                            </div>
> > +                            <span class="punctuation">/</span>
> > +                            <div class="label-with-selector">
> > +                              <label class="multiselector-label">Image Stream</label>
> > +                              <select
> > +                                class="form-control multiselector select-builder-name"
> > +                                ng-model="options.pickedBuildFromImageStream"
> > +                                ng-options="name as name for name in buildFrom.imageStreams track by name"
> > +                                ng-change="updateTags(options.pickedBuildFromImageStream, true)">
> > +                              </select>
>
> You should be able to type in one that doesn't exist
>
> What did we decide about that during the design review? I can't remember.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6528/files#r48876185>.
>